Git-commit: 227d5adbd22c6711f0db0d8ea83c5b65ff39486c
From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Mon, 8 Mar 2021 09:56:29 +0800
Subject: [PATCH] arm64: ipi_nmi: fix compile error when CONFIG_KGDB is
 disabled
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Patch-mainline: Queued in openEuler repo, version 5.10.0-4.9.0
References: openEuler-21.03
Git-repo: https://gitee.com/openeuler/kernel.git

hulk inclusion
category: bugfix
bugzilla: 50485
CVE: NA

-------------------------------------------------------------------------

Fix the following compile error.

arch/arm64/kernel/ipi_nmi.c: In function ‘ipi_nmi_handler’:
arch/arm64/kernel/ipi_nmi.c:54:7: error: implicit declaration of function ‘kgdb_nmicallback’ [-Werror=implicit-function-declaration]
  if (!kgdb_nmicallback(cpu, get_irq_regs()))
       ^~~~~~~~~~~~~~~~

Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 arch/arm64/kernel/ipi_nmi.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/kernel/ipi_nmi.c b/arch/arm64/kernel/ipi_nmi.c
index 898d69c37429..3b105852fc17 100644
--- a/arch/arm64/kernel/ipi_nmi.c
+++ b/arch/arm64/kernel/ipi_nmi.c
@@ -46,13 +46,14 @@ bool arch_trigger_cpumask_backtrace(const cpumask_t *mask, bool exclude_self)
 static irqreturn_t ipi_nmi_handler(int irq, void *data)
 {
 	irqreturn_t ret = IRQ_NONE;
-	unsigned int cpu = smp_processor_id();
 
 	if (nmi_cpu_backtrace(get_irq_regs()))
 		ret = IRQ_HANDLED;
 
-	if (!kgdb_nmicallback(cpu, get_irq_regs()))
+#ifdef CONFIG_KGDB
+	if (!kgdb_nmicallback(smp_processor_id(), get_irq_regs()))
 		ret = IRQ_HANDLED;
+#endif
 
 	return ret;
 }
-- 
2.31.1

